client/src/components/RegistrationForm/RegistrationForm.tsx

```diff
@@ -0,0 +1,155 @@
+        }
+        const submittable = this.isSubmittable() &&
+                            error_password === null &&
+                            error_email === null &&
+                            error_birthday === null;
+
+        let action = <RaisedButton disabled={!submittable}
+                                    type='submit'
+                                    style={{marginTop: '20px'}}
+                                    label='Register' />;
+
+        if (this.props.pending) {
+            action = <CircularProgress style={{paddingTop: '20px'}}/>;
+        }
+
+        let errors: any = null;
+        if (this.props.errors) {
+            errors = this.props.errors.map((error: string, idx: number) => {
+                return <div key={'error-' + idx} style={{color: 'red'}}>{error}<br /></div>;
+            });
+        }
+
+        return (
+            <form onSubmit={this.submit.bind(this)}>
+                {errors}
+                <TextField hintText='Email' name='user'
+                            floatingLabelText='Email'
+                            value={this.state['user']}
+                            onChange={this.bindValue.bind(this)}
+                            errorText={error_email}
+                            disabled={this.props.pending} />
+                <br />
+
+                <TextField hintText='Full Name' name='name'
+                            floatingLabelText='Full Name'
+                            value={this.state['name']}
+                            onChange={this.bindValue.bind(this)}
+                            disabled={this.props.pending} />
+                <br />
+
+                <TextField hintText='Password' name='password1'
+                            floatingLabelText='Password'
+                            type='password' value={this.state['password1']}
+                            onChange={this.bindValue.bind(this)}
+                            errorText={error_password}
+                            disabled={this.props.pending} />
+                <br />
+
+                <TextField hintText='Confirm Password' name='password2'
+                            type='password' value={this.state['password2']}
+                            onChange={this.bindValue.bind(this)}
+                            errorText={error_password}
+                            disabled={this.props.pending} />
+                <br />
+                <TextField floatingLabelText='Birthday' name='birthday'
+                            hintText='mm/dd/yyyy'
+                            value={this.state['birthday']}
+                            onChange={this.formatBirthday.bind(this)}
+                            errorText={error_birthday}
+                            disabled={this.props.pending} />
+                <br />
+                {action}
+            </form>
+        );
+    }
+
+    submit(ev: Event) {
+        const email = this.state['user'];
```

Comment:
This is fine but maybe we don't need to pull them all out of state first? I'm ambivalent on this one just bringing it up since it may have just been a holdover. Whatever you prefer is fine.